tools/c7n_azure/c7n_azure/resources/storage.py

```diff
@@ -465,3 +465,44 @@ def update_logging(storage_type, storage_account, logging_settings, session=None
 
         return getattr(client, 'set_{}_service_properties'
                        .format(storage_type))(logging=logging_settings)
+
+
+@Storage.action_registry.register('set-https-traffic')
+class SetHttpsTrafficAction(AzureBaseAction):
+    """Action that updates the EnableHttpsTrafficOnly setting on Storage Accounts.
+
+    :example:
+
+        Set EnableHttpsTrafficOnly to True
+
+    .. code-block:: yaml
+
+        policies:
+            - name: set-https-traffic-to-true
+            resource: azure.storage
+            actions:
+                - type: set-https-traffic
+                value: true
+    """
+
+    # Default to true assuming user wants secure connection
+    schema = type_schema(
+        'set-https-traffic',
+        required=[],
+        **{
+            'value': {'type': 'boolean', "default": True},
+        })
+
+    def __init__(self, data, manager=None):
+        super(SetHttpsTrafficAction, self).__init__(data, manager)
+        self.log = logging.getLogger('custodian.azure.storage')
```

Comment:
So this was commented on in another pr, logs are meant to be singletons and there is already a log instance on the base action class as a class attribute